tests/unit/local_transport.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright 2022 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include <chrono>
+#include <thread>
+
+#include "gtest/gtest.h"
+
+#include "io/local_transport/local_system.hpp"
+#include "io/local_transport/local_transport.hpp"
+#include "io/transport.hpp"
+
+using memgraph::io::Address;
+using memgraph::io::Io;
+using memgraph::io::ResponseFuture;
+using memgraph::io::ResponseResult;
+using memgraph::io::local_transport::LocalSystem;
+using memgraph::io::local_transport::LocalTransport;
+
+struct CounterRequest {
+  uint64_t proposal;
+};
+
+struct CounterResponse {
+  uint64_t highest_seen;
+};
+
+void run_server(Io<LocalTransport> io) {
+  uint64_t highest_seen = 0;
+
+  while (!io.ShouldShutDown()) {
+    std::cout << "[SERVER] Is receiving..." << std::endl;
+    auto request_result = io.Receive<CounterRequest>();
+    if (request_result.HasError()) {
+      std::cout << "[SERVER] Error, continue" << std::endl;
```

Comment:
I've replaced all cout calls in the test with spdlog::info. In this case, it's an expected error because the server will occasionally time out so that it can handle periodic functionality.